examples/swarm/docker-compose-pg.yml

```diff
@@ -0,0 +1,86 @@
+version: '3.4'
+
+secrets:
+  pgsql:
+    file: ./etc/secrets/pgsql
+  pgsql_repl:
+    file: ./etc/secrets/pgsql_repl
+
+services:
+  sentinel:
+    image: sorintlab/master-pg10:latest
+    command: gosu stolon stolon-sentinel --cluster-name stolon-cluster --store-backend=etcdv3 --store-endpoints http://etcd-00:2379,http://etcd-01:2379,http://etcd-02:2379 --log-level debug
+    networks:
+      - etcd_etcd
+      - pgdb
+    deploy:
+      replicas: 2
+      update_config:
+        parallelism: 1
+        delay: 30s
+        order: stop-first
+        failure_action: pause
+
+  keeper1:
+    image: sorintlab/master-pg10:latest
+    hostname: keeper1
+    environment:
+      - PGDATA=/var/lib/postgresql/data
+    volumes:
+      - pgkeeper1:/var/lib/postgresql/data
+    secrets:
+      - pgsql
+      - pgsql_repl
+    command: gosu stolon stolon-keeper --pg-listen-address keeper1 --pg-repl-username replication --uid keeper1 --pg-su-username postgres --pg-su-passwordfile /run/secrets/pgsql --pg-repl-passwordfile /run/secrets/pgsql_repl --data-dir /var/lib/postgresql/data --cluster-name stolon-cluster --store-backend=etcdv3 --store-endpoints http://etcd-00:2379,http://etcd-01:2379,http://etcd-02:2379 --log-level debug
+    networks:
+      - etcd_etcd
+      - pgdb
+    deploy:
+      replicas: 1
+
+  keeper2:
+    image: sorintlab/master-pg10:latest
+    hostname: keeper2
+    environment:
+      - PGDATA=/var/lib/postgresql/data
+    volumes:
+      - pgkeeper2:/var/lib/postgresql/data
+    secrets:
+      - pgsql
+    command: gosu stolon stolon-keeper --pg-listen-address keeper2 --pg-repl-username replication --uid keeper2 --pg-su-username postgres --pg-su-passwordfile /run/secrets/pgsql --pg-repl-passwordfile /run/secrets/pgsql --data-dir /var/lib/postgresql/data --cluster-name stolon-cluster --store-backend=etcdv3 --store-endpoints http://etcd-00:2379,http://etcd-01:2379,http://etcd-02:2379 --log-level debug
+    networks:
+      - etcd_etcd
+      - pgdb
+    deploy:
+      replicas: 1
+
+  proxy:
+    image: sorintlab/master-pg10:latest
+    command: gosu stolon stolon-proxy --listen-address 0.0.0.0 --cluster-name stolon-cluster --store-backend=etcdv3 --store-endpoints http://etcd-00:2379,http://etcd-01:2379,http://etcd-02:2379 --log-level info
+    networks:
+      - etcd_etcd
+      - pgdb
+    ports:
+      - 5432:5432
+    deploy:
+      replicas: 2
+      update_config:
+        parallelism: 1
+        delay: 30s
+        order: stop-first
+        failure_action: rollback
+
+volumes:
+  pgkeeper1:
+    driver: local
```

Comment:
Does using the local volume plugin will create some problems since on multiple node cluster since the keeper cannot be started on another node than the one where the volume has been created? Perhaps some notes to change this to a real persistent volume will be good.

Reply:
You're right, this section deserves probably more explanations. 
Out of the box, only the local driver is available with docker swarm. That's the reason why this swarm example is based on the local driver. 
I specifically chose to provide different names for volumes of keeper1 and keeper2 so that we can't have any volume mismatch if keeper1 and keeper2 are deployed on the same node of the swarm.
Nevertheless, IMHO, it makes no sense to deploy all keepers on the same node. So I would recommend to set some constraints on both keepers so that each node would have its own keeper. Here is an example of of a configuration where keeper1 will be deployed only on a node whose nodename label is equal to node1:
```
  keeper1:
    deploy:
      placement:
        constraints: [node.labels.nodename == node1]
     ...
```
In case each keeper is deployed on its own node, a node failure should not prevent the cluster to remain in function.

I'm not sure if my explanations are clear to you. I would propose to add a comment in the README to explain this approach. @sgotti What do you think?

Of course, if a specific docker volume plugin is available that could enable to share volumes between nodes, the volume section of the docker-compose-pg.yml would have to be adapted accordingly to leverage this plugin and there's no need to set any constraints.

Reply:
@Nowheresly Your explanation is right and I agree with you. A comment in the README will be great!

Reply:
Thanks for the review

Reply:
@sgotti I updated the README to explain how to set labels on each node and how to set constraints on each keeper. Hopefully, it should make sense to swarm users.